pkg/controllers/certificatestore_controller.go

```diff
@@ -81,26 +84,18 @@ func (r *CertificateStoreReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, err
 	}
 
-	// fetch certificates based on the provider
-	switch certStore.Spec.Provider {
-	case "azurekeyvault":
-		certificates, err := azurekeyvault.GetCertificates(ctx, attributes)
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("Error fetching certificates in store %v with azure key vault provider, error: %w", resource, err)
-		}
-		certificatesMap[resource] = certificates
-		logger.Infof("%v certificates fetched for certificate store %v", len(certificates), resource)
-	case "inline":
-		certificates, err := inline.GetCertificates(ctx, attributes)
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("Error fetching certificate in store %v with inline provider, error: %w", resource, err)
-		}
-		certificatesMap[resource] = certificates
-		logger.Infof("%v certificates fetched for certificate store %v", len(certificates), resource)
-	default:
+	providers := certificateprovider.GetCertificateProviders()
+	provider, registered := providers[certStore.Spec.Provider]
```

Comment:
not a strong opinion, just wonder if we want to trim and lower case the `certStore.Spec.Provider`.

Reply:
added TrimSpaceAndToLower, thanks!